dv/cosim/spike_cosim.cc

```diff
@@ -24,8 +25,10 @@ SpikeCosim::SpikeCosim(const std::string &isa_string, uint32_t start_pc,
     log_file = log->get();
   }
 
+  isa_parser = new isa_parser_t(isa_string.c_str(), "MU");
```

Comment:
I am not sure if this is the "correct" way of doing this. I've tried using a `unique_ptr` type for this but if I understand correctly, this doesn't work because we need to pass `isa_parser *` to constructor of `processor` (part of Spike), casting doesn't work and I think it also shouldn't work here. Anyway, some input on this is appreciated.

Reply:
There are two clean ways to do this.

First (and most similar to what you've got), turn `isa_parser` into a `unique_ptr<isa_parser_t>` and use `isa_parser.get()` to get the underlying pointer.

Second (and probably less effort), make `isa_parser` an actual `isa_parser_t` as a field and call its constructor in the class initialisation list. Then you can pass it to the `processor_t` constructor with just `&isa_parser`.

Reply:
Thanks Rupert, I've now switched to the first option :-)